R/outpack_misc.R

```diff
@@ -44,9 +44,19 @@ git_info <- function(path) {
   if (is.null(repo)) {
     return(NULL)
   }
-  list(sha = gert::git_commit_id(repo = repo),
-       branch = gert::git_branch(repo = repo),
-       url = gert::git_remote_list(repo = repo)$url)
+
+  sha <- tryCatch(gert::git_commit_id(repo = repo),
+                  error = function(e) NA)
+
+  branch <- gert::git_branch(repo = repo)
+  if (is.null(branch) || identical(branch, "HEAD")) {
+    # NULL can be returned when working in a repo that has no commits yet.
```

Comment:
I wonder if it's worth having sentinel values for these - could be `(empty)` for no commits and `(detached)` for detached head, or tracking this though something in the metadata? Probably not worth getting too wound up about though